packages/sessions-sdk-react/src/session-limits-tab.tsx

```diff
@@ -0,0 +1,143 @@
+import { useState, useRef, useEffect } from "react";
+
+import styles from "./session-limits-tab.module.css";
+import { SessionLimits } from "./session-limits.js";
+import type { EstablishedSessionState } from "./session-provider.js";
+import { useSessionContext } from "./session-provider.js";
+import { Spinner } from "./spinner.js";
+import {
+  StateType as TokenDataStateType,
+  useTokenAccountData,
+} from "./use-token-account-data.js";
+
+const ONE_SECOND_IN_MS = 1000;
+const ONE_MINUTE_IN_MS = 60 * ONE_SECOND_IN_MS;
+const ONE_HOUR_IN_MS = 60 * ONE_MINUTE_IN_MS;
+const ONE_DAY_IN_MS = 24 * ONE_HOUR_IN_MS;
+
+export const SessionLimitsTab = ({
+  sessionState,
+}: {
+  sessionState: EstablishedSessionState;
+}) => (
+  <div className={styles.sessionLimitsTab}>
+    <SessionExpiryBanner expiration={sessionState.expiration} />
+    <SessionLimitsForm sessionState={sessionState} />
+  </div>
+);
+
+const SessionLimitsForm = ({
+  sessionState,
+}: {
+  sessionState: EstablishedSessionState;
+}) => {
+  const state = useTokenAccountData(sessionState);
+  const { whitelistedTokens, enableUnlimited } = useSessionContext();
+
+  switch (state.type) {
+    case TokenDataStateType.Error:
+    case TokenDataStateType.Loaded: {
+      return (
+        <SessionLimits
+          header={
+            <div className={styles.header}>
+              <h2 className={styles.title}>Session Limits</h2>
+              <span className={styles.description}>
+                Limit how many tokens this app is allowed to interact with
+              </span>
+            </div>
+          }
+          className={styles.sessionLimits}
+          bodyClassName={styles.body}
+          footerClassName={styles.footer}
+          tokens={whitelistedTokens}
+          hideCancel
+          initialLimits={
+            new Map(
+              state.type === TokenDataStateType.Error
+                ? undefined
+                : state.data.sessionLimits.map(({ mint, sessionLimit }) => [
+                    mint,
+                    sessionLimit,
+                  ]),
+            )
+          }
+          onSubmit={
+            "updateSession" in sessionState
+              ? sessionState.updateSession
+              : undefined
+          }
+          buttonText="Update Limits"
```

Comment:
<img width="507" height="328" alt="Screenshot 2025-09-30 at 1 10 56 PM" src="https://github.com/user-attachments/assets/7cf83821-921f-441c-871b-b15fa5306c5b" />
lot of 1password getting in the way. not sure if anything we can do but just fyi